Blog GSoC 2022 (#220)
MDAnalysis/MDAnalysis.github.io: _posts/2022-03-07-gsoc2022.md

---
layout: post
title: Google Summer of Code 2022
---

<p>
<img
src="https://developers.google.com/open-source/gsoc/resources/downloads/GSoC-Vertical.svg"
title="Google Summer of Code 2022" alt="Google Summer of Code with
MDAnalysis 2022"
style="float: right; height: 10em; " />
</p>

> MDAnalysis has been accepted as an [organization][org] into [Google Summer of Code 2022][gsoc]!

Reviewer: ```suggestion
MDAnalysis has been accepted as an [organization][org] for [Google Summer of Code 2022][gsoc]!
```
into reads strangely, but that could just be me

> If you are a student who is interested in working with us

Reviewer: ```suggestion
If you are a new open source contributor who is interested in working with us
```

this summer, please read the advice and links below and write to us on the
[mailing list].

We are looking forward to all applications. Note that the GSoC program was
[updated](https://opensource.googleblog.com/2021/11/expanding-google-summer-of-code-in-2022.html)
compared to previous years: GSoC will welcome not just students, but *any new
and beginner open source contributors over 18 years old*. Projects are also now
scoped as either 175-hour (medium) or 350-hour (long) size. Finally, the duration
can be extended from the standard 12 weeks to 22 weeks.

The application window **deadline** is **April 19, 2022 18:00 (UTC)**. As
part of the application process you must familiarize yourself with [Google
Summer of Code 2022][gsoc]. 

If you are interested in working with us please read on and contact us on our

> [mailing list].  [Apply] as soon as possible; the application window opens on

Reviewer: ```suggestion
[mailing list]. [Apply] as soon as possible; the application window opens on
```
Double space?

April 4, 2022 but potential GSoC contributors are expected to familiarize
themselves with application requirements and mentoring organizations as soon as
possible. It's also never too early to discuss application ideas with us!

# Project Ideas

If you have your own idea about a potential project we'd love to work with you
to develop this idea; please write to us on the [developer list][mailing list]
to discuss it there.

We also have listed several [possible projects][ideas] for you to work on. Our
initial list of ideas (see summaries in the table below) contains various
projects of different scope and with different skill requirements. However,

> check the [ideas][] page — we might add more ideas after the posting date of

Reviewer: Maybe we shouldn't have the table here and just get them to go look at the ideas page? It seems a bit redundant to post it in two separate places?

Author: I'd leave the table because it acts as an advertisement and shows a bit more what we are thinking about — even to people who are not going to apply.

this post.

> You don't need to have all the skills that we are listing, although that helps,
> of course. But you need to demonstrate to us that you're able and keen to learn

Reviewer: ```suggestion
The listed skills are suggested rather than essential, although they will be used
as part of our decision criteria in choosing GSoC contributors. Our only essential
requirement is that you need to demonstrate to us that you're able and keen to learn
```

Reviewer: Is this fully true? I got the feeling from folks that some of these really should be non-optional skills (note, I'm happy with making them non-optional).

| 3       | [Extend MDAnalysis Interoperability]({{ site.github.wiki }}/GSoC-2022-Project-Ideas#project-3-extend-interoperability)                  | medium     | 350 hours   | Extend converters module to other relevant packages                                            | Python, Molecular Modelling                      | @lilyminium, @IAlibay, @fiona-naughton, @hmacdope  |
| 4       | [Benchmarking and performance optimization]({{ site.github.wiki }}/GSoC-2022-Project-Ideas#project-4-benchmarking-and-performance-optimization)                  | medium     | 175 hours   |  write benchmarks for automated performance analysis and address performance bottlenecks | Python                      | @hmacdope, @orbeckst, @jbarnoud  |
| 5       | [Context-aware guessers]({{ site.github.wiki }}/GSoC-2022-Project-Ideas#project-5-context-aware-guessers) | medium | 350 hours   | Extend how the library guesses properties such as bonds, masses or atom symbols; and write guessers that know about the context of the system (database of origin, force field...) | Python, Molecular modelling | @jbarnoud @micaela-matta @IAlibay |

> # Information for Students

Reviewer: For here and everywhere else that says "students".

Due to the recent changes, GSoC no longer calls folks that take part solely "students" but also "contributors".

From the mailing list:

"With this change we needed a new way to address the participants who are not students so we came up with the term “GSoC Contributors”​​ as it is more welcoming than participants. We are trying to use the full term “GSoC Contributor” so they aren’t confused in a general conversation about contributors already in your community. "

It would be good for us to switch to "GSoC Contributors" as our main wording from now on?

Author: yes, thanks for reading carefully

> # Information for Students

Reviewer: ```suggestion
# Information for prospective GSoC Contributors
```

> You must meet our [own requirements] if you want to be a student with MDAnalysis

Reviewer: ```suggestion
You must meet our [own requirements] if you want to be a GSoC Contributor with MDAnalysis
```

this year (read all the docs behind these links!). You must also meet the
[eligibility criteria]. Our [GSoC FAQ][] collects common questions

> from students.

Reviewer: ```suggestion
from GSoC Contributors.
```

The MDAnalysis community values diversity and is committed to
providing a productive, harassment-free to every member. Our [Code of
Conduct] explains the values that we as a community uphold. *Every

> member (and every GSoC student) agrees to follow the [Code of

Reviewer: ```suggestion
member (and every GSoC Contributor) agrees to follow the [Code of
```

Conduct]*.

As a start to get familiar with MDAnalysis and open source development you
should follow these steps:

## Complete the Quick Start Guide

We have a [Quick Start Guide] explaining the basics of MDAnalysis. You
should go through it at least once to understand how MDAnalysis is
used. Continue reading the [User Guide] to learn more.

## Introduce yourself to us

> Introduce yourself on the [mailing list]. Tell us what you plan to work

Reviewer: ```suggestion
Introduce yourself on the [mailing list]. Tell us your github handle, what you plan to work
```
Or something along these lines, linking folks from their email to their PR is always difficult.

on during the summer or what you have already done with MDAnalysis.

## Close an issue of MDAnalysis

You must have *at least one commit in the development branch of MDAnalysis* in
order to be eligible, i.e. you must demonstrate that you have been seriously
engaged with the MDAnalysis project.  We have a list of [easy bugs][] and
suggested [GSOC Starter issues][GSOC Starter] to work on in our issue tracker
on GitHub. *We only accept one [GSOC Starter issue][GSOC Starter] per

> applicant* so that everybody gets a chance. If you want to dive deeper, tackle
> some of the other issues.

Reviewer: ```suggestion
applicant* so that everybody gets a chance. If you want to dive deeper, we
encourage you to tackle some of the other issues in our issue tracker.
```

> We also appreciate if you write more tests or update/improve our documentation.

Reviewer: ```suggestion
We also appreciate contributions which add more tests or update/improve our documentation.
```

We recommend you start your application by working on an issue. It will give
you a better understanding of MDAnalysis as a project and improve the quality
of your application.

To start developing for MDAnalysis have a look at our
guide on [contributing to MDAnalysis][dev-guide] and write to us on
the [mailing list][] if you have more questions about setting up a
development environment or how to contribute.

— Hugo @hmacdope, Micaela @micaela-matta, Jonathan @jbarnoud, Richard @richardjgowers, Lily @lilyminium, Fiona @fiona-naughton, Irfan @IAlibay, Oliver @orbeckst

> [org]: https://summerofcode.withgoogle.com/organizations/

Author: Need to add link to our org page.

> [org]: https://summerofcode.withgoogle.com/organizations/

Reviewer: ```suggestion
[org]: https://summerofcode.withgoogle.com/programs/2022/organizations/mdanalysis
```

> [Apply]: https://summerofcode.withgoogle.com/student-signup/

Author: 404 at the moment

Reviewer: Unless one of us is going to remember to change this on April 4th, should we just not have the link?

Author: I am removing the deep link and just put the raw site URL into the text.

[Code of Conduct]: {{ site.baseurl }}/pages/conduct/
[eligibility criteria]: https://developers.google.com/open-source/gsoc/faq#what_are_the_eligibility_requirements_for_participation
[own requirements]: {{ site.github.wiki }}/Google-Summer-Of-Code#our-expectations-from-students
[easy bugs]: https://github.com/MDAnalysis/mdanalysis/issues?q=is%3Aopen+is%3Aissue+label%3ADifficulty-easy
[GSOC Starter]: https://github.com/MDAnalysis/mdanalysis/issues?q=is%3Aopen+is%3Aissue+gsoc+label%3A%22GSOC+Starter%22
[Quick Start Guide]: https://userguide.mdanalysis.org/stable/quickstart.html
[User Guide]: https://userguide.mdanalysis.org/
[ideas]: {{ site.github.wiki }}/GSoC-2022-Project-Ideas
[gsoc]: https://summerofcode.withgoogle.com/

> [dev-guide]: https://www.mdanalysis.org/UserGuide/contributing.html

Reviewer: ```suggestion
[dev-guide]: https://userguide.mdanalysis.org/stable/contributing.html
```

redirects to the same thing, but for whomever copies this next year in case the redirect one day goes away.

title="Google Summer of Code 2022" alt="Google Summer of Code with
MDAnalysis 2022"
style="float: right; height: 10em; " />
</p>

MDAnalysis has been accepted as an [organization][org] for [Google Summer of
Code 2022][gsoc]!  If you are interested in working with us this summer and you
are new to open source, please read the advice and links below and write to us
on the [mailing list].

We are looking forward to all applications. Note that the GSoC program was
[updated](https://opensource.googleblog.com/2021/11/expanding-google-summer-of-code-in-2022.html)
compared to previous years: GSoC will welcome not just students, but *any new
and beginner open source contributors over 18 years old*. Projects are also now
scoped as either 175-hour (medium) or 350-hour (long) size. Finally, the duration
can be extended from the standard 12 weeks to 22 weeks.

The application window **deadline** is **April 19, 2022 18:00 (UTC)**. As
part of the application process you must familiarize yourself with [Google
Summer of Code 2022][gsoc]. 

If you are interested in working with us please read on and contact us

> on our [mailing list].  Apply at as soon as possible at

Reviewer 2: ```suggestion
on our [mailing list].  Apply as soon as possible at
```
